drivers/ukbus/pci/arch/x86_64/pci_bus.c

```diff
@@ -94,9 +152,15 @@ static inline int pci_driver_add_device(struct pci_driver *drv,
 	config_addr = (PCI_ENABLE_BIT)
 			| (addr->bus << PCI_BUS_SHIFT)
 			| (addr->devid << PCI_DEVICE_SHIFT);
+	dev->config_addr = config_addr;
 	PCI_CONF_READ(__u16, &dev->base, config_addr, IOBAR);
 	PCI_CONF_READ(__u8, &dev->irq, config_addr, IRQ);
 
+	if (arch_pci_find_cap(dev, PCI_CAP_MSIX, &cap_offset) == 0)
```

Comment:
Can you add some comment regarding this code? I find it difficult to follow the change without a comment explaining the reason behind.